projectroles/app_settings.py

```diff
@@ -101,37 +107,35 @@ class AppSettingAPI:
     @classmethod
     def _check_project_and_user(cls, scope, project, user):
         """
-        Ensure one of the project and user parameters is set.
+        Ensure project and user parameters are set according to scope.
 
-        :param scope: Scope of Setting (USER, PROJECT, PROJECT_USER)
+        :param scope: Scope of Setting (USER, PROJECT, PROJECT_USER, SITE)
         :param project: Project object
         :param user: User object
-        :raise: ValueError if none or both objects exist
-        """
-        if scope == APP_SETTING_SCOPE_PROJECT:
-            if not project:
-                raise ValueError('Project unset for setting with project scope')
-            if user:
-                raise ValueError('User set for setting with project scope')
-        elif scope == APP_SETTING_SCOPE_USER:
-            if project:
-                raise ValueError('Project set for setting with user scope')
-            if not user:
-                raise ValueError('User unset for setting with user scope')
-        elif scope == APP_SETTING_SCOPE_PROJECT_USER:
-            if not project:
-                raise ValueError(
-                    'Project unset for setting with project_user scope'
-                )
-            if not user:
-                raise ValueError(
-                    'User unset for setting with project_user scope'
-                )
-        elif scope == APP_SETTING_SCOPE_SITE:
-            if project:
-                raise ValueError('Project set for setting with site scope')
-            if user:
-                raise ValueError('User set for setting with site scope')
+        :raise: ValueError if parameters are not valid for scope
+        """
+        valid_scope = APP_SETTING_VALID_SCOPE_VALIDATION.get(scope)
```

Comment:
My original idea was to simplify this even further. We can just do:

```
if not APP_SETTING_SCOPE_ARGS[scope] == {'project': project is not None, 'user': user is not None}:
    raise ValueError()  # etc etc
```

Also, you should rebase to dev and call this also in your newly created `delete_by_scope()` method :)